File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        vector<string> words;
        string temp = "";
        for (char i: sentence)
        {
            if (i == ' ')
            {
                words.push_back(temp);
                temp = "";
            }
            else{
                temp += i;
            }
                
        }
        words.push_back(temp);

        int l = 0;
        for (int i = 0; i < words.size(); i++){
            for(int j = 0; j < searchWord.length(); j++){
                if (words[i][j] != searchWord[j])
                    break;
                if (j == searchWord.length() - 1)
                    return i + 1;
            }
        }
        return -1;
    }
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        words = sentence.split(' ')
        for i in range(len(words)):
            for j in range(len(words[i])):
                if words[i][j] != searchWord[j]:
                    break
                if j == len(searchWord) -1 :
                    return i + 1
        return -1
            
        
